File: Projeto_5/src/professor.py
from usuario import Usuario

# criação da classe Professor
class  Professor(Usuario):
    def __init__(self, nome: str, email: str, area_de_atuacao: str, disciplinas: list):
        super().__init__(nome, email, "Professor")
        self.area_de_atuacao = area_de_atuacao
        self.disciplinas = disciplinas

    def gerar_relatorio(self):
        return f"Professor: {self.nome}, Área de Atuação: {self.area_de_atuacao}, Disciplinas: {', '.join(self.disciplinas)}"

File: Projeto_5/src/professor.cpp
#include "../include/professor.h"
#include <iostream>

// Construtor padrão
Professor::Professor() : Usuario("", "", "Professor"), areaDeAtuacao("") {}

// Construtor parametrizado
Professor::Professor(const std::string& nome, const std::string& email, const std::string& tipo, const std::string& areaDeAtuacao)
    : Usuario(nome, email, "Professor"), areaDeAtuacao(areaDeAtuacao) {}

// Método para adicionar uma disciplina ministrada
void Professor::adicionarDisciplina(const std::string& disciplina) {
    disciplinasMinistradas.push_back(disciplina);
}

// Sobrescrita de gerarRelatorio
void Professor::gerarRelatorio() const {
    std::cout << "Professor: " << nome << "\n"
              << "Area de Atuacao: " << areaDeAtuacao << "\n"
              << "Disciplinas Ministradas:\n";
    for (const auto& disciplina : disciplinasMinistradas) {
        std::cout << "  - " << disciplina << "\n";
    }
}
